api-get-object.go

```diff
@@ -289,6 +291,21 @@ func (o *Object) doGetRequest(request getRequest) (getResponse, error) {
 		return response, response.Error
 	}
 
+	// Check if ETag has changed since last request.
+	switch {
+	// If response doesn't carry a valid ETag, ignore.
+	case response.objectInfo.ETag == "":
+
+	// Set Object's etag on first request.
+	case o.etag == "":
+		o.etag = response.objectInfo.ETag
+
+	// Check if etag from response matches Object's etag.
+	case o.etag != response.objectInfo.ETag:
+		return response, errors.New("object has been modified since last read")
```

Comment:
@harshavardhana Since this error is returned by `minio.Object`'s `Stat`, `Read` and `ReadAt`, I was thinking if `syscall.EBADFD` would be more appropriate?